src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/BuildPropertyPage/DefineConstantsValueProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    /// Supports the setting of constants in the property page, corresponding to the "DefineConstants" property
+    /// </summary>
+    [ExportInterceptingPropertyValueProvider("DefineConstants", ExportInterceptingPropertyValueProviderFile.ProjectFile)]
+    internal sealed class DefineConstantsValueProvider : InterceptingPropertyValueProviderBase
```

Comment:
I forgot to add, you'll need to add an [AppliesTo(ProjectCapabilities.VisualBasic)] here, because we have a _different_ property with the same name for non-VB projects

Reply:
so this should also be renamed 😄